iris_grib/grib_phenom_translation.py

```diff
@@ -320,3 +321,70 @@ def cf_phenom_to_grib2_info(standard_name, long_name=None):
     if standard_name is not None:
         long_name = None
     return _CF_GRIB2_TABLE[(standard_name, long_name)]
+
+
+class GribCode(namedtuple('GribCode',
+                          'edition discipline category number')):
+    """
+    An object representing a specific Grib phenomenon identity.
+
+    Basically a namedtuple of (edition, discipline, category, number).
+
+    Also provides a string representation, and supports creation from: another
+    similar object; a tuple of numbers; or any string with 4 separate decimal
+    numbers in it.
+
+    """
+    __slots__ = ()
+
+    def __new__(cls, edition_or_string,
+                discipline=None, category=None, number=None):
+        args = (edition_or_string, discipline, category, number)
+        nargs = sum(arg is not None for arg in args)
+        if nargs == 1:
+            # Single argument: convert to a string and extract 4 integers.
+            # NOTE: this also allows input from a GribCode, or a plain tuple.
+            edition_or_string = str(edition_or_string)
+            edition, discipline, category, number = \
+                cls._fournums_from_gribcode_string(edition_or_string)
+        elif nargs == 4:
+            edition = edition_or_string
+            edition, discipline, category, number = [
+                int(arg)
+                for arg in (edition, discipline, category, number)]
+        else:
+            msg = ('Cannot create GribCode from {} arguments, '
+                   '"GribCode{!r}" : '
+                   'expected either 1 or 4 non-None arguments.')
+            raise ValueError(msg.format(nargs, args))
+
+        return super(GribCode, cls).__new__(
+            cls, edition, discipline, category, number)
+
+    RE_PARSE_FOURNUMS = re.compile(4 * r'[^\d]*(\d*)')
+
+    @classmethod
+    def _fournums_from_gribcode_string(cls, edcn_string):
+        parsed_ok = False
+        nums_match = cls.RE_PARSE_FOURNUMS.match(edcn_string).groups()
+        if nums_match is not None:
+            try:
+                nums = [int(grp) for grp in nums_match]
+                parsed_ok = True
+            except ValueError:
+                pass
+
+        if not parsed_ok:
+            msg = ('Invalid argument for GribCode creation, '
+                   '"GribCode({!r})" : '
+                   'requires 4 numbers, separated by non-numerals.')
+            raise ValueError(msg.format(edcn_string))
+
+        return nums
+
+    PRINT_FORMAT = 'GRIB{:1d}:d{:d}.c{:d}.n{:d}'
```

Comment:
Rather than
` 'GRIB2:d1.c1.n1`
How about
`GRIB2:d001c001n001`
or
 `GRIB2:d1c1n1`

Reply:
Will fix to follow https://github.com/SciTools/iris-grib/pull/156#issuecomment-560362457 suggestions